modules/standard/CTypes.chpl

```diff
@@ -954,6 +956,25 @@ module CTypes {
     }
     return c_pointer_return(b.buff[0]);
   }
+
+  @chpldoc.nodoc
+  inline proc c_ptrTo_helper(ref c: class): c_ptr(void)
+  {
+    if (isUnmanagedClass(c)) {
+      return __primitive("cast", c_ptr(void), c);
+    } else {
+      return __primitive("cast", c_ptr(void), c.borrow());
+    }
+  }
+  @chpldoc.nodoc
+  inline proc c_ptrTo_helper(ref c: class?): c_ptr(void)
+  {
+    if (isUnmanagedClass(c)) {
+      return __primitive("cast", c_ptr(void), c);
+    } else {
+      return __primitive("cast", c_ptr(void), c.borrow());
+    }
+  }
```

Comment:
I think you can get rid of the branch? You should be able to call `.borrow()` on an `unmanaged`

Reply:
Oh, not sure why I was assuming you can't.